paperqa/types.py

```diff
@@ -591,32 +598,43 @@ def populate_bibtex_key_citation(  # noqa: PLR0912
                     entries={data["key"]: new_entry}
                 ).to_string("bibtex")
                 # clear out the citation, since it will be regenerated
-                if data.get("overwrite_fields_from_metadata", True):
+                if data.get("overwrite_citation_from_metadata", True):
                     data["citation"] = None
             except Exception:
                 logger.warning(
                     "Failed to generate bibtex for"
                     f" {data.get('docname') or data.get('citation')}"
                 )
-        if not data.get("citation") and data.get("bibtex") is not None:
+        if data.get("citation") is None and data.get("bibtex") is not None:
             data["citation"] = format_bibtex(
                 data["bibtex"], missing_replacements=CITATION_FALLBACK_DATA  # type: ignore[arg-type]
             )
-        elif not data.get("citation"):
+        elif data.get("citation") is None:
             data["citation"] = data.get("title") or CITATION_FALLBACK_DATA["title"]
         return data
 
     @model_validator(mode="before")
     @classmethod
     def validate_all_fields(cls, data: Mapping[str, Any]) -> dict[str, Any]:
+
         data = deepcopy(data)  # Avoid mutating input
         data = dict(data)
+
+        if (
+            isinstance(data.get("overwrite_citation_from_metadata"), str)
+            and data.get("overwrite_citation_from_metadata", "").lower()
+            in ENV_VAR_MISMATCH
+        ):
+            data |= {"overwrite_citation_from_metadata": False}
+
+        data = cls.lowercase_doi_and_populate_doc_id(data)
         data = cls.lowercase_doi_and_populate_doc_id(data)
         data = cls.remove_invalid_authors(data)
         data = cls.misc_string_cleaning(data)
         data = cls.inject_clean_doi_url_into_data(data)
         data = cls.add_preprint_journal_from_doi_if_missing(data)
         data = cls.populate_bibtex_key_citation(data)
+
```

Comment:
[PR nit] try to be conservative when adding one line whitespaces, it adds extra diff and the whitespace can break locality

If you do like this whitespace, feel free to keep it, more just sharing is all